### ISSUE TITLE
[completion] - TypeError: Cannot create property 'filterText' on string '$'

### ISSUE THREAD

**Issue by FongHou** [neoclide/coc.nvim#2250]
## Result from CocInfo

```
   1 vim version: NVIM v0.4.3
   2 node version: v14.5.0
   3 coc.nvim version: 0.0.78-bdd9a9e140
   4 coc.nvim directory: /mnt/c/Users/houfe/haskeller/.config/nvim/plugged/coc.nvim
   5 term: tmux-256color
   6 platform: linux
```
## Describe the bug

```
2020-08-15T23:32:37.433 DEBUG (pid:25287) [completion] - trigger completion with {
  word: 'd/',
  bufnr: 4,
  col: 1,
  synname: 'clojureRainbow_lv0_p0',
  filepath: '/mnt/c/Users/houfe/haskeller/dev/cljdata/src/cljdata/datalog.clj',
  blacklist: [],
  line: '(d/)',
  filetype: 'clojure',
  linenr: 29,
  input: 'd/',
  colnr: 4,
  changedtick: 11,
  triggerCharacter: '/'
}
2020-08-15T23:32:37.489 DEBUG (pid:25287) [completion-complete] - Source "around" takes 3ms
2020-08-15T23:32:37.489 DEBUG (pid:25287) [completion-complete] - Source "buffer" takes 2ms
2020-08-15T23:32:37.489 DEBUG (pid:25287) [completion-complete] - Source "tabnine" takes 3ms
2020-08-15T23:32:37.490 DEBUG (pid:25287) [completion-complete] - Source "snippets" takes 4ms
2020-08-15T23:32:37.490 DEBUG (pid:25287) [completion-complete] - Source "file" takes 2ms
2020-08-15T23:32:37.495 DEBUG (pid:25287) [completion-complete] - Source "tag" takes 6ms
2020-08-15T23:32:37.510 DEBUG (pid:25287) [completion-complete] - Source "conjure" takes 21ms
2020-08-15T23:32:37.550 DEBUG (pid:25287) [completion-complete] - Source "languageserver.clojure-lsp-1" takes 64ms
2020-08-15T23:32:37.550 INFO (pid:25287) [completion-complete] - Results from: around,buffer,conjure,languageserver.clojure-lsp-1
2020-08-15T23:32:37.550 ERROR (pid:25287) [completion] - TypeError: Cannot create property 'filterText' on string '$'
    at Complete.filterResults (/mnt/c/Users/houfe/haskeller/.config/nvim/plugged/coc.nvim/build/index.js:70278:33)
    at Complete.doComplete (/mnt/c/Users/houfe/haskeller/.config/nvim/plugged/coc.nvim/build/index.js:70408:21)
    at async Completion._doComplete (/mnt/c/Users/houfe/haskeller/.config/nvim/plugged/coc.nvim/build/index.js:40621:21)
    at async Completion.startCompletion (/mnt/c/Users/houfe/haskeller/.config/nvim/plugged/coc.nvim/build/index.js:40505:13)
    at async Completion.triggerCompletion (/mnt/c/Users/houfe/haskeller/.config/nvim/plugged/coc.nvim/build/index.js:40749:9)
    at async Completion.onTextChangedI (/mnt/c/Users/houfe/haskeller/.config/nvim/plugged/coc.nvim/build/index.js:40688:13)
2020-08-15T23:32:37.890 DEBUG (pid:25287) [events] - Event: CursorHoldI [ 4 ]
```

## Reproduce the bug

**We will close your issue when you don't provide minimal vimrc and we can't
reproduce it**

Appears only with clojure-lsp

```
  "languageserver": {
     "clojure-lsp": {
           "command": "bash",
           "args": ["-c", "clojure-lsp"],
           "filetypes": ["clojure"],
           "rootPatterns": ["project.clj"],
           "additionalSchemes": ["jar", "zipfile"],
           "trace.server": "verbose",
           "initializationOptions": { "keep-require-at-start?": true }
     }
}

```
- Create file `mini.vim` with：

  ```vim
  set nocompatible
  set runtimepath^=/path/to/coc.nvim
  filetype plugin indent on
  syntax on
  set hidden
  ```

- Start (neo)vim with command: `vim -u mini.vim`

- Operate vim.

**Comment by chemzqm** [comment]
Should be problem of your language server, checkout https://github.com/neoclide/coc.nvim/wiki/Debug-language-server#using-output-channel

**Comment by FongHou** [comment]
I checked language server output per instruction. Result from textDocument/completion response is an array of objects. Nothing looks suspicious. 

```
[
    {
         "label": "d",
         "data": "d"
     },
    ...
]

```
 
Mapping the stacktrace line number 70278 in coc log to the source line [here](https://github.com/neoclide/coc.nvim/blob/63b52c1463c4b4ce3e907347a5e3c7ffe6e4bcf0/src/completion/complete.ts#L194). The error message "TypeError: Cannot create property 'filterText' on string '$'" seems saying the value of item var on that line is not an object, but a string instead, which I think it was pushed on items array in this.result somewhere on coc client side. I did some code search where completion item objects were created and pushed, but was not able to narrow down the exact code path. 

Hope these additional detail would help...
